modAionImpl/src/org/aion/zero/impl/blockchain/AionBlockchainImpl.java

```diff
@@ -907,15 +907,15 @@ public synchronized FastImportResult tryFastImport(final Block block) {
     public static boolean reachedFullSync = false;
 
     public ImportResult tryToConnect(final Block block) {
+        checkKernelShutdownForCLI();
         return tryToConnect(new BlockWrapper(block));
     }
 
     public synchronized ImportResult tryToConnect(final BlockWrapper blockWrapper) {
-        checkKernelShutdownForCLI();
         return tryToConnectWithTimedExecution(blockWrapper).getLeft();
     }
 
-    private void checkKernelShutdownForCLI() {
+    public void checkKernelShutdownForCLI() {
```

Comment:
it would be safer if this method would be called with the block number as parameter instead of relying on the `bestBlock` variable